app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,52 @@
+class CommentsController < ApplicationController
+  load_and_authorize_resource
+
+  # issue_comments POST /issues/:issue_id/comments(.:format) comments#create
+  def create
+    @resource = Issue.find(params[:issue_id])
+    authorize! :read, @resource
+
+    @comment.commentable = @resource
+    @comment = Comment.new if @comment.save
+    @pagy, @comments = pagy(@resource.comments)
+    # TODO: Add a proper decorator class for this type of thing
+    @pagy.instance_variable_set(:@custom_link, project_issue_path(@resource.project, @resource))
+    respond_to :js
+  end
+
+  # edit_comment GET /comments/:id/edit(.:format) comments#edit
+  def edit
+    respond_to :js
+  end
+
+  # comment PATCH  /comments/:id(.:format) comments#update
+  # comment PUT    /comments/:id(.:format) comments#update
+  def update
+    respond_to do |format|
+      if @comment.update(comment_params)
+        format.js { render 'show' }
+      else
+        format.js { render 'edit' }
+      end
+    end
+  end
+
+  # comment DELETE /comments/:id(.:format) comments#destroy
+  def destroy
+    respond_to do |format|
+      if @comment.destroy
+        flash[:comment_deletion_success] = t('comments.successful_deletion_message')
```

Comment:
use :notice for success, and :error for errors

Reply:
Can there be more than 1 notifications for a page? If so then that thing will not work i.e. using :notice for notice etc

Reply:
If there is a problem for like notification color we can compare like '/success$/ =~ key.to_s'

Reply:
should I use :comment_deletion_notice ?

Reply:
Personally I think there should be an info message (:info or :notice), a success message (:success) and an error message (:error) and maybe a warning one (:warning)